### DISCUSSION TITLE
TTS reading unwanted asterisks and incorrect example in librechat.example.yaml

### DISCUSSION THREAD

**Discussion by bfogels1** [danny-avila/LibreChat#3655] [Feature Requests & Suggestions]
### What happened?

TTS reads out many "asterisks" for bolded words. It might improve user experience to simply remove those asterisks. This could be implemented in  Librechat\librechat\api\server\services\Files\Audio\TTSService.js line 267 by modifying input prior to calling the strategy. 

Additionally, when implementing the speech settings in librechat.yaml by following librechat.example.yaml, there was an error when the key voice: 'alloy' was used. Instead the field expects  'voices' which is a string. 

To clarify confusion, I would update line 34 of librechat.example.yaml to voices: ['']

### Steps to Reproduce

the following does not work:

speech:
  tts:
    openai:
      url: 'http://host.docker.internal:8010/v1/audio/speech'
      apiKey: ''
      model: 'tts-1'
      voice: 'alloy'
      
 but the following works:

speech:
  tts:
    openai:
      url: 'http://host.docker.internal:8010/v1/audio/speech'
      apiKey: ''
      model: 'tts-1'
      voices: ['alloy', 'echo', 'fable', 'nova', 'onyx']

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
These are not bugs, rather suggestions:

> TTS reads out many "asterisks" for bolded words. It might improve user experience to simply remove those asterisks. This could be implemented in Librechat\librechat\api\server\services\Files\Audio\TTSService.js line 267 by modifying input prior to calling the strategy.

By default, all text, symbols or not, is read out loud. This can be improved as you're suggesting, but needs to take into account proper syntax/grammar read-aloud for all languages, not just English.

> Additionally, when implementing the speech settings in librechat.yaml by following librechat.example.yaml, there was an error when the key voice: 'alloy' was used. Instead the field expects 'voices' which is a string.

Right. If you want one voice you should use this format:

`voices: ['alloy']`

**Reply by bfogels1** [reply]
I think that the "*" character is universally used across languages for bolding due to markdown. Simply removing the asterisk character made for a much better user experience in my case.

Librechat\librechat\api\server\services\Files\Audio\TTSService.js line 267

// Remove all asterisk (*) characters
input = input.replace(/\*/g, '');

As for the example, I agree a single voice would be voices: ['alloy'], but that is not what is in the example config file.

Thanks for all your hard work with this project, I am absolutely loving using it!

**Reply by danny-avila** [reply]
Thanks for taking the time to write the initial report as well as this response!

Yes, I think we can start with the asterisks for a better UX.

@berry-13 noticed after your message that the example should be updated, and made a commit here: https://github.com/danny-avila/LibreChat/pull/3657

**Reply by berry-13** [reply]
This is already planned. I have already prepared a function that cleans up all unwanted words for the STT whisper. I could also do this for backticks and these particular symbols

**Reply by JustinApt** [reply]
> This is already planned. I have already prepared a function that cleans up all unwanted words for the STT whisper. I could also do this for backticks and these particular symbols

Hi, I'm running into this issue as well. 

Powered with Kokoro (af-heart), the TTS response reads out all the asterisks, hashes or other Makrdown indicators. 

Is the fix (cleaning up unwanted words) still planned?

I'm really enjoying using Librechat and appreciate all of you for making this possible :)
